### ISSUE TITLE
Fix S1144: False positive on async Main

### ISSUE THREAD

**Issue by nnpcYvIVl** [SonarSource/sonar-csharp#888]
A false positive S1144 is thrown on the line: 
```csharp 
static async Task Main(string[] args)
```

In C# 7.1 this is a valid entrypoint method. https://blogs.msdn.microsoft.com/mazhou/2017/05/30/c-7-series-part-2-async-main/

```csharp
using System;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class Program
    {
        static async Task Main(string[] args)
        {
            Console.WriteLine("Test");
        }
    }
}
```

**Comment by valhristov** [comment]
Hi @nnpcYvIVl, thanks for the feedback, we will fix this issue once we start working on C# 7 in the near future.

**Comment by RowlandShaw** [comment]
I've started seeing this again with version 4.5.0.3786

**Comment by valhristov** [comment]
Hi @RowlandShaw, is the project you are seeing this error connected to a SonarQube instance? If yes, even with the latest SonarLint, you could be using the analyzers which are installed on SonarQube.

Could you please check if you have SonarAnalyzer.CSharp nuget package referenced in your project. If yes, what's its version?

If you are not connected, or if you are using the latest version, could you please send us a simple reproducer application, because we are unable to see issues with the code snippet in the original ticket.

**Comment by RowlandShaw** [comment]
One thing I did notice was making the class static resolved the issue

**Comment by valhristov** [comment]
Hi @RowlandShaw, I am unable to reproduce the problem with the latest version of SonarC#, could you please open a new issue and provide a simple code snippet that triggers the rule? Thank you in advance!
